File: clojure/rotational-cipher/src/rotational_cipher.clj
(ns rotational-cipher)

(defn rotate-char [shift]
  (fn [x]
    (let [a (int \a)
          z (int \z)
          A (int \A)
          Z (int \Z)
          num (int x)
          low (<= a num z)
          cap (<= A num Z)
          base (if low a A)
          shifted (- num base)]
    (if
      (or low cap)
      (char
         (+ base (mod (+ shift shifted) 26)))
      x))))

(defn rotate [text shift]
  (apply
    str
    (map
      (rotate-char shift)
      (seq text))))

File: python/rotational-cipher/rotational_cipher.py
def rotate(text, key):
    return ''.join([__rotate_chr(i, key) for i in text])

UPPER_A = ord('A')
UPPER_Z = ord('Z')
LOWER_A = ord('a')
LOWER_Z = ord('z')

def __rotate_chr(char, key):
    char_ord = ord(char)

    if (UPPER_A <= char_ord <= UPPER_Z):
        base = UPPER_A
    elif (LOWER_A <= char_ord <= LOWER_Z):
        base = LOWER_A
    else:
        return char

    shifted = char_ord - base
    return chr((shifted + key) % 26 + base)
